modules/core/include/opencv2/core/hal/intrin.hpp

```diff
@@ -1226,6 +1226,105 @@ namespace CV__SIMD_NAMESPACE {
 //! @endcond
 
 
+//! @name Exponential
+//! @{
+//! @brief Calculates exponent of every vector element
+    inline v_float32 v_exp(const v_float32 &x) {
+        const v_float32 _vexp_lo_f32 = vx_setall_f32(-88.3762626647949f);
+        const v_float32 _vexp_hi_f32 = vx_setall_f32(89.f);
+        const v_float32 _vexp_half_fp32 = vx_setall_f32(0.5f);
+        const v_float32 _vexp_one_fp32 = vx_setall_f32(1.f);
+        const v_float32 _vexp_LOG2EF_f32 = vx_setall_f32(1.44269504088896341f);
```

Comment:
1.
My suggestion is to move this and similar implementations into a separate file.
There is already `simd_utils.impl.hpp`.
It could be reused or created a new one like `simd_math.impl.hpp`

2.
Also all functions should be guarded if some ISA has better support for these functions.
At least intrin_cpp.hpp could use `std::exp()` as reference (to ensure that test code is adequate)

Reply:
1. The constants will only used in the `v_exp` function currently. Maybe some of them will be used in other functions like `v_sin` and `v_cos`? Should I move them to a separate file as global variables? I just want them to be created when they are used for the first time.
2. There is already `OPENCV_HAL_IMPL_MATH_FUNC(v_exp, std::exp, _Tp)` in line 728 of file `intrin_cpp.hpp `

Reply:
Let's implement item 2 too. I want to benchmark both options before merge.

Reply:
Also I propose to add reference to some paper, description, or implementation details, if you have it.

Reply:
> Also I propose to add reference to some paper, description, or implementation details, if you have it.

I cannot find the original paper; I only found the code and some comments. I already added these comments and references.

Reply:
The fallback with just loop and standatrd exp is still needed. We have platforms, where fma/sse/avx/neon are not implemented. Looped exp should be better there.